gamemodes/terrortown/gamemode/client/cl_target_data.lua

```diff
@@ -0,0 +1,164 @@
+---
+-- @author Mineotopia
+
+TARGET_DATA = {}
+
+---
+-- Binds two target data tables to the @{TARGET_DATA} object
+-- @param table data The data table about the focused entity
+-- @param table params The default table with the params that should be modified by the hook
+-- @return @{TARGET_DATA} The object to be used in the hook
+-- @internal
+-- @realm client
+function TARGET_DATA:BindTarget(data, params)
+	self.data = data
+	self.params = params
+
+	return self
+end
+
+---
+-- Returns the currently focused entity
+-- @return Entity The focused entity
+-- @realm client
+function TARGET_DATA:GetEntity()
+	return self.data.ent
+end
+
+---
+-- Returns the distance to the focused entity
+-- @return number The distance to the focused entity
+-- @realm client
+function TARGET_DATA:GetEntityDistance()
+	return self.data.distance
+end
+
+---
+-- Enables/Disables the outline around a focused entity
+-- @param [default=true] boolean enb_outline A boolean defining the outline state
+-- @realm client
+function TARGET_DATA:EnableOutline(enb_outline)
+	self.params.drawOutline = enb_outline or true
+end
+
+---
+-- Enables/Disables the targetID text and icons
+-- @param [default=true] boolean enb_text A boolean defining the text stat
```

Comment:
*`state`

Reply:
Could you pls explain why needing this function?

Reply:
In case someone wants only an outline and no text

Reply:
and why not just leaving the text empty?

Reply:
Because it is implemented this way. It was the same since targetID got implemented. I thought it would be the most straightforward way to enable/disable targetID

Especially if one addon adds the text, another addon can disable it without having to bother with the raw data.

Thinking about it, I should change it probably in a way that targetID isn't displayed once one addon sets it to false. What do you think about this?

Reply:
If any add-on wanna prevent it (returning false), this prevention should count. imho.

Reply:
yep, I've already changed it yesterday